include/glow/Support/Error.h

```diff
@@ -45,6 +39,78 @@ template <typename> struct IsLLVMExpected : public std::false_type {};
 template <typename T>
 struct IsLLVMExpected<llvm::Expected<T>> : public std::true_type {};
 
+/// Represents errors in Glow. GlowErr track the file name and line number of
+/// where they were created as well as a textual message and/or a error code to
+/// help identify the type of error the occurred programtically.
+class GlowErr final : public llvm::ErrorInfo<GlowErr> {
+public:
+  /// Used by ErrorInfo::classID.
+  static char ID;
+  /// An enumeration of error codes representing various possible errors that
+  /// could occur.
+  enum EC {
+    // An unknown error ocurred. This is the default value.
+    UNKNOWN = 0,
+    // Model loader encountered an unsupported shape.
+    MODEL_LOADER_UNSUPPORTED_SHAPE = 1,
+    // Model loader encountered an unsupported operator.
+    MODEL_LOADER_UNSUPPORTED_OPERATOR = 2,
+    // Model loader encountered an unsupported attribute.
+    MODEL_LOADER_UNSUPPORTED_ATTRIBUTE = 3,
+    // Model loader encountered an unsupported datatype.
+    MODEL_LOADER_UNSUPPORTED_DATATYPE = 4,
+    // Model loader encountered an unsupported ONNX version.
+    MODEL_LOADER_UNSUPPORTED_ONNX_VERSION = 5,
+    // Model loader encountered an invalid protobuf.
+    MODEL_LOADER_INVALID_PROTOBUF = 6,
+  };
+
+  /// GlowErr is not convertable to std::error_code. This is included for
+  /// compatiblity with ErrorInfo.
+  virtual std::error_code convertToErrorCode() const override {
+    return llvm::inconvertibleErrorCode();
+  }
+
+  /// Log to \p OS relevant error information including the file name and
+  /// line number the GlowErr was created on as well as the message and/or error
+  /// code the GlowErr was created with.
+  void log(llvm::raw_ostream &OS) const override {
+    OS << "file: " << fileName_ << " line: " << lineNumber_;
+    if (ec_ != EC::UNKNOWN) {
+      OS << " error code: " << ec_;
+    }
+    if (!message_.empty()) {
+      OS << " message: " << message_;
+    }
+  }
+
+  GlowErr(llvm::StringRef fileName, int32_t lineNumber, llvm::StringRef message,
```

Comment:
`lineNumber` is `int32_t`, while `lineNumber_` is `size_t`.

Reply:
dang, thanks!